### PR TITLE
Decouple HIveThriftServer2 and SparkSQLClientDriver

### PR DESCRIPTION
**Pull request by zhixingheyi-tian**
## What changes were proposed in this pull request?

Decouple HIveThriftServer2 and SparkSQLClientDriver introduced from Spark.

To solve #1003 

## How was this patch tested?

unit tests, integration tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @ivoson

**Comment by LuciferYang** [comment]
Great Work @zhixingheyi-tian

File: src/main/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala [hunk 1]
```diff
@@ -34,50 +34,11 @@ import org.apache.spark.util.Utils
...
-
-      sparkContext.ui.foreach(new OapTab(_))
-      initialized = true
-    }
-  }
-
-  // This is to enable certain OAP features, like UI, in non-Spark SQL CLI/ThriftServer conditions
+  // This is to enable certain OAP features, like UI, even
+  // in non-Spark SQL CLI/ThriftServer conditions
   def initWithoutCreatingOapSession(): Unit = synchronized {
```
Comment (LuciferYang) [inline src/main/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala]: Maybe we should rename this method to `initWithoutCreatingSparkSession`?

Comment (zhixingheyi-tian) [reply src/main/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala]: @LuciferYang 
Hi,
 I used to want to rename this method name, but initWithoutCreatingOapSession is also called by OapDriverRuntime, which will be useful in Spark-shell scenarios

Comment (LuciferYang) [reply src/main/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala]: Maybe `OapSession`  in method `initWithoutCreatingOapSession`  is specifically to `OapSession.scala` which we deleted in old version, but I am not sure.

Comment (LuciferYang) [reply src/main/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala]: but it's a trivial problem

Comment (zhixingheyi-tian) [reply src/main/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala]: @LuciferYang 

you are right.

Comment (zhixingheyi-tian) [reply src/main/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala]: Done

**Comment by LuciferYang** [comment]
@zhixingheyi-tian merge to master first , thx for ur work ~

**Pull request merged**